boards: fix dependency for LV_COLOR_SWAP_16 to use configdefault

danieldegrasse opened the pull request:
Use configdefault when enabling LV_COLOR_SWAP_16 within boards and
shield definitions, to avoid OR'ing the dependencies for the Kconfig
symbol. Otherwise, a user manually selecting LV_COLOR_DEPTH will
encounter build errors as LV_COLOR_SWAP_16 may be enabled when
LV_COLOR_DEPTH_16 is not selected

Fixes #81546

danieldegrasse commented:
@faxe1008 do you feel this is the right fix? Or should the LVGL transparency sample not be setting `CONFIG_LV_COLOR_DEPTH_32` directly?

faxe1008 reviewed (commented):
Hmm wouldn't the easier fix be adding the depedency to the stub symbol within `modules/lvgl/Kconfig`?

EDIT:// yeah you are right, this wont suffice since they are ORed. Regarding the sample, selecting the depth makes sense since otherwise the sample is kinda pointless.

erwango commented on boards/adi/max32662evkit/Kconfig.defconfig:
```diff
@@ -20,7 +20,7 @@ choice LV_COLOR_DEPTH
 endchoice
 
 config LV_COLOR_16_SWAP
-	default y
+	default y if LV_COLOR_DEPTH_16
```
what about using [`configdefault`](https://docs.zephyrproject.org/latest/build/kconfig/extensions.html#kconfig-extensions) ?

faxe1008 replied:
Very good call, forgot that this exists

danieldegrasse replied:
Wasn't aware of this- I'll update the PR to use this, thanks!

faxe1008 reviewed (changes requested):
See @erwango comment

danieldegrasse committed:
boards: fix dependency for LV_COLOR_SWAP_16 to use configdefault

Use configdefault when enabling LV_COLOR_SWAP_16 within boards and
shield definitions, to avoid OR'ing the dependencies for the Kconfig
symbol. Otherwise, a user manually selecting LV_COLOR_DEPTH will
encounter build errors as LV_COLOR_SWAP_16 may be enabled when
LV_COLOR_DEPTH_16 is not selected

Fixes #81546

Signed-off-by: Daniel DeGrasse <daniel.degrasse@nxp.com>

danieldegrasse commented:
> See @erwango comment

Updated

danieldegrasse commented:
#81611 should fix the other test failure blocking this PR
